torch/_higher_order_ops/associative_scan.py

```diff
@@ -79,32 +109,150 @@ def add(x: torch.Tensor, y: torch.Tensor):
 
     if not torch._dynamo.is_compiling():
         with _set_compilation_env(), torch._dynamo.utils.disable_cache_limit():
-            return torch.compile(associative_scan, fullgraph=True)(
-                combine_fn, input, dim
-            )
+            if host_side:
+                return torch.compile(associative_scan_host_side, fullgraph=True)(
+                    combine_fn, input, dim, reverse
+                )
+                # return associative_scan_host_side(combine_fn, input, dim, False)
+            else:
+                if reverse:
+                    raise ValueError(
+                        "For the device-side associative scan, the reverse flag has not yet been implemented"
+                    )
```

Comment:
We could do two flips here as well to support it.

Reply:
I did this in my clean-up as well. One might though need to change the interface of `torch.cumsum` and `torch.cumprod`  to include that parameter, I guess?